Replicator/Pusher.cc

```diff
@@ -650,6 +658,18 @@ namespace litecore { namespace repl {
                 || !_pushingDocs.empty()
                 || _revisionBytesAwaitingReply > 0) {
             level = kC4Busy;
+        } else if (!_revsToRetry.empty()) {
+            logInfo("%d documents failed to push and will be retried", (int)_pendingSequences.size());
+            level = kC4Busy;
+            Pusher* pusher = const_cast<Pusher *>(this);
+            pusher->_caughtUp = false;
+            auto revsToRetry = move(_revsToRetry);
+            pusher->_revsToRetry.clear();
+            for(const auto& revToRetry : revsToRetry) {
+                pusher->_pushingDocs[revToRetry->docID] = revToRetry;
+            }
+            
+            pusher->gotChanges(make_shared<RevToSendList>(revsToRetry), _maxPushedSequence, {});
```

Comment:
not much context, but function name suggest`computeActivityLevel` is meant to compute the activity level and no retry handling. is it possible to separate this method out?

Reply:
I can make it into another method but this function is called often from many different places and so it's not like I can bring it "up one level" or something like that.